cmd/pctl/search.go

```diff
@@ -25,41 +25,43 @@ func searchCmd() *cli.Command {
 				Value:       "table",
 				Usage:       "Output format. json|table",
 			},
+			&cli.BoolFlag{
+				Name:    "all",
+				Aliases: []string{"a"},
+				Usage:   "Search all available profiles",
+			},
 		},
 		Action: func(c *cli.Context) error {
-			searchName, catalogClient, err := parseArgs(c)
-			if err != nil {
-				_ = cli.ShowCommandHelp(c, "search")
-				return err
-			}
-
-			profiles, err := catalog.Search(catalogClient, searchName)
-			if err != nil {
-				return err
-			}
+			var profiles []profilesv1.ProfileCatalogEntry
 			outFormat := c.String("output")
-			if outFormat == "table" {
-				if len(profiles) == 0 {
-					fmt.Printf("No profiles found matching: '%s'\n", searchName)
-					return nil
-				}
-			}
-
-			var f formatter.Formatter
-			f = formatter.NewTableFormatter()
-			getter := searchDataFunc(profiles)
 
-			if outFormat == "json" {
-				f = formatter.NewJSONFormatter()
-				getter = func() interface{} { return profiles }
+			if c.Bool("all") {
+				catalogClient, err := getClient(c)
+				if err != nil {
+					_ = cli.ShowCommandHelp(c, "search")
+					return err
+				}
+				profiles, err = catalog.Search(catalogClient, "all")
+				if err != nil {
+					return err
+				}
+			} else {
+				searchName, catalogClient, err := parseArgs(c)
+				if err != nil {
+					_ = cli.ShowCommandHelp(c, "search")
+					return err
+				}
+				profiles, err = catalog.Search(catalogClient, searchName)
+				if err != nil {
+					return err
+				}
 			}
 
-			out, err := f.Format(getter)
-			if err != nil {
-				return err
+			outErr := outputFormatter(profiles, outFormat)
+			if outErr != nil {
+				return outErr
 			}
 
-			fmt.Println(out)
 			return nil
```

Comment:
I think we can just `return outputFormatter(profiles, outFormat)` here too.

Can we rename this func too? With `outputFormatter` I feel like I should be getting an object back. What do you think of just `formatOutput` or `printOutput`?

Reply:
done , renamed it to `formatOutput` 👍🏻